pkg/client/rp/relying_party.go

```diff
@@ -157,6 +165,13 @@ func (rp *relyingParty) ErrorHandler() func(http.ResponseWriter, *http.Request,
 	return rp.errorHandler
 }
 
+func (rp *relyingParty) UnauthorizedHandler() func(http.ResponseWriter, *http.Request, string, string) {
+	if rp.unauthorizedHandler == nil {
+		rp.unauthorizedHandler = DefaultUnauthorizedHandler
+	}
+	return rp.unauthorizedHandler
```

Comment:
race condition on the RP. Better to handle the population of the `rp.unauthorizedHandler` once in the constructor.

Reply:
Hm, I do copy and paste from existing function. The race conditions goes to the both functions, too?

https://github.com/zitadel/oidc/blob/2b35eeb83560d69c6b492eb5dcad02ad30c6da55/pkg/client/rp/relying_party.go#L146-L158